src/vzlogger.cpp

```diff
@@ -549,6 +583,13 @@ int main(int argc, char *argv[]) {
 					::sleep(1); // 1s should be ok. will introduce a shutdown latency >1s
 				}
 			}
+			if (mainLoopReopenLogfile) {
+				mainLoopReopenLogfile = false;
+				print(log_info, "closing logfile for re-opening (requested with SIGUSR1)", "");
+				closeLogFile();
+				openLogfile();
```

Comment:
there is a small race cond here where new data is lost, or?
A function like "closeAndOpen" keeping the lock would be safer. (Could internally use the same functions close/openLogfile just without those locking themselves?

Reply:
is this critical?
i was hoping to avoid that mess of having to provide those functions with and without locking :\

Reply:
well I wouldn't really mind. Not really sure (didn't check) whether the startup log logic would kick in here. Up to you. I wouldn't block this PR due to this.

Reply:
pushed a fix, but it's ugly